### PR TITLE
Fix clippy for 1.80

### PR DESCRIPTION
**Pull request by cjpatton**
Also, make sure we're building the benches.

### CODE REVIEW

**Review by RageKnify — APPROVED** [review]

File: crates/daphne/benches/pine.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ fn pine(c: &mut Criterion) {
         (100_000, 320 * 6),
         (1_000_000, 1_000 * 8),
     ] {
-        let pine = Pine::new_64(1.0, dimension, 15, chunk_len).unwrap();
+        let pine = Pine::new_64(1 << 15, dimension, 15, chunk_len).unwrap();
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply crates/daphne/benches/pine.rs:20]: Actually, this has to do with the change from `f64` to `u64` in https://github.com/cloudflare/daphne/pull/636. In more detail, the L2 norm bound is a real number that we encode in a particular way into an integer by multiplying by the 2^frac_bits, which here is 15.

File: .github/workflows/daphneci.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ jobs:
         run: cargo fmt --all --check
       - name: Documentation
         run: RUSTDOCFLAGS='-D warnings' cargo doc --locked --no-deps --all-features --workspace
+      - name: Build
+        run: cargo build --all-targets --all-features
```
Comment (mendess) [inline .github/workflows/daphneci.yml:36]: why add this?

Comment (cjpatton) [reply .github/workflows/daphneci.yml:36]: CI was not building the benchmarks and did not detect a build error in daphne/benches/pine.rs.

Comment (mendess) [reply .github/workflows/daphneci.yml:36]: can you do only `--all-targets`? `--all-features` can cause weird issues where features that shouldn't be enabled together are enabled together

Comment (cjpatton) [reply .github/workflows/daphneci.yml]: Done.

**Review by mendess — APPROVED** [review]

**Pull request merged**